File: Bit Soccer/Bit Soccer.py
def solve(performance_indexes, team_performance_index):
    def get_zero_indexes(number):
        result = []
        index_counter = 0
        while number != 0:
            if number % 2 == 0:
                result.append(index_counter)
            number = number >> 1
            index_counter += 1
        return result

    def is_valid(indexes, number):
        for index in zero_indexes:
            if (number >> index) % 2 == 1:
                return False
        return True

    result = 0
    zero_indexes = get_zero_indexes(team_performance_index)

    for zero_index in performance_indexes:
        if zero_index <= team_performance_index and is_valid(zero_indexes, zero_index):
            result = result | zero_index

        if result == team_performance_index:
            return 'YES'

    return 'NO'


def main():
    number_of_players = int(input())
    performance_indices = list(map(int, input().split()))
    number_of_queries = int(input())
    for _ in range(number_of_queries):
        team_performance_index = int(input())
        result = solve(performance_indices, team_performance_index)
        print(result)


if __name__ == '__main__':
    main()

File: Bit Soccer/Bit Soccer.cpp
#include <iostream>

using namespace std;

bool isPosible(long long int *, long long int, int);
bool isValid(long long int, long long int);

int main(int argc, char const *argv[])
{
    int n, q;
    long long int g;

    cin >> n;
    long long int p[n];

    for (int i = 0; i < n; i++)
        cin >> p[i];

    cin >> q;

    for (int i = 0; i < q; i++)
    {
        cin >> g;
        if (isPosible(p, g, n))
            cout << "YES" << endl;
        else
            cout << "NO" << endl;
    }
    return 0;
}

bool isPosible(long long int *p, long long int q, int size)
{
    long long int result = 0;
    for (int i = 0; i < size; i++)
    {
        if (isValid(q, p[i]))
        {
            result |= p[i];
        }
    }
    return result == q;
}

bool isValid(long long int base, long long int n)
{
    bool result = true;
    while (n != 0)
    {
        if (base % 2 == 0 && n % 2 == 1)
        {
            result = false;
            break;
        }
        base >>= 1;
        n >>= 1;
    }
    return result;
}